### PR TITLE
refactor Project entity and add Project entity test

### PR DESCRIPTION
**Pull request by VincentETN**
Two files update:
1. In Project.java, I replaced those get and set functions with @Getter and @Setter, lombok annotation. (Test all pass)
2. I added ProjectTest.java, which is used to test functions of Project.

### CODE REVIEW

**Comment by Xanonymous-GitHub** [comment]
Suggest not to use the main branch to initiate a pull request.
See [this article](https://guides.github.com/introduction/flow/) for more details.

File: src/test/java/pvs/app/entity/ProjectTest.java [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+        repositorySet = new HashSet<>();
+        repositorySet.add(githubRepository);
+        project.setRepositorySet(repositorySet);
+    }
+
+    @Test
+    public void getNameTest() {
+        assertEquals("myProject", project.getName());
+    }
+}
```
Comment (Xanonymous-GitHub) [inline src/test/java/pvs/app/entity/ProjectTest.java:42]: Suggest adding a newline in the end of the file.
```suggestion
}

```

File: src/test/java/pvs/app/entity/ProjectTest.java [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+public class ProjectTest {
+    Project project;
+    Repository githubRepository;
+    Set<Repository> repositorySet;
+
+    @Before
+    public void setup() {
+        project = new Project();
+        project.setMemberId(1L);
+        project.setName("myProject");
```
Comment (Xanonymous-GitHub) [inline src/test/java/pvs/app/entity/ProjectTest.java:26]: In order to ensure that the expected project name is the name of the test project, Suggest use a constant variable to declare this name and use it on both `project.setName` and `assertEquals`.
```java
final String testProjectName = "myProject";
```
```suggestion
        project.setName(testProjectName);
```

File: src/test/java/pvs/app/entity/ProjectTest.java [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+        githubRepository.setRepositoryId(1L);
+
+        repositorySet = new HashSet<>();
+        repositorySet.add(githubRepository);
+        project.setRepositorySet(repositorySet);
+    }
+
+    @Test
+    public void getNameTest() {
+        assertEquals("myProject", project.getName());
```
Comment (Xanonymous-GitHub) [inline src/test/java/pvs/app/entity/ProjectTest.java:40]:
```suggestion
        assertEquals(testProjectName, project.getName());
```

File: src/test/java/pvs/app/entity/ProjectTest.java [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+    @Before
+    public void setup() {
+        project = new Project();
+        project.setMemberId(1L);
+        project.setName("myProject");
+
+        githubRepository = new Repository();
+        githubRepository.setType("github");
+        githubRepository.setUrl("https://github.com/facebook/react");
+        githubRepository.setRepositoryId(1L);
```
Comment (Xanonymous-GitHub) [inline src/test/java/pvs/app/entity/ProjectTest.java:31]: Since your unit test only tests whether the expected project name can be obtained, there is no need to set the type and id of the test project, so maybe we can remove those lines.

File: src/test/java/pvs/app/entity/ProjectTest.java [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+
+import java.util.HashSet;
+import java.util.Set;
+
+import static org.junit.Assert.assertEquals;
+
+@RunWith(SpringRunner.class)
+@SpringBootTest(classes = Application.class)
+public class ProjectTest {
+    Project project;
```
Comment (Xanonymous-GitHub) [inline src/test/java/pvs/app/entity/ProjectTest.java:18]: Suggest use a more-detailed name to this.

```suggestion
    Project testProject;
```

File: src/test/java/pvs/app/entity/ProjectTest.java [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+        githubRepository.setUrl("https://github.com/facebook/react");
+        githubRepository.setRepositoryId(1L);
+
+        repositorySet = new HashSet<>();
+        repositorySet.add(githubRepository);
+        project.setRepositorySet(repositorySet);
+    }
+
+    @Test
+    public void getNameTest() {
```
Comment (Xanonymous-GitHub) [inline src/test/java/pvs/app/entity/ProjectTest.java:39]: In my opinion, the reason why you want to write this unit test should be to test whether the project name is correctly set, not to test whether the set project name can be correctly "obtained", so suggest to change the function name for convey more precise meaning.

```suggestion
    public void expectCorrectlySetName() {
```

**Comment by Xanonymous-GitHub** [comment]
Also suggest rename this PR, because the word `project` is easy to misunderstand other meanings, like github ... etc.
Maybe you can change to `refactor Project entity and add Project entity test` ...

**Comment by Xanonymous-GitHub** [comment]
Before I started a review, there exist two same-name commit, makes me go into ponder, what are these commits doing? Why they have same message? Why the message is equal to this PR's name... XDDDD

So maybe the commit message `refactor Project and add Project test` can be change to a more clear expression~~

File: src/test/java/pvs/app/entity/ProjectTest.java (hunk 6 above)
Comment (VincentETN) [reply src/test/java/pvs/app/entity/ProjectTest.java:39]: 因為我想新增是否被刪除的屬性，這屬性有預設值，所以我先想到測getter

File: src/main/resources/application.yml [hunk 7]
```diff
@@ -25,7 +25,7 @@ spring:
   datasource:
     url: jdbc:postgresql://localhost:5432/PVS
     username: postgres
-    password: root
+    password: SE2021f
```
Comment (Xanonymous-GitHub) [inline src/main/resources/application.yml:28]: Suggest revert this change.

Comment (VincentETN) [reply src/main/resources/application.yml:28]: ok

**Comment by Xanonymous-GitHub** [comment]
please notify us after this PR is ready for review and merge.

File: src/main/java/pvs/app/entity/Project.java [hunk 8]
```diff
@@ -29,43 +34,7 @@
...
-    public Set<Repository> getRepositorySet() {
-        return repositorySet;
-    }
-
-    public void setRepositorySet(Set<Repository> repositorySet) {
-        this.repositorySet = repositorySet;
-    }
+    // 0: deleted; 1: normal
+    @Column(columnDefinition = "integer default 1")
+    private Integer status;
```
Comment (Xanonymous-GitHub) [inline src/main/java/pvs/app/entity/Project.java:39]:
```suggestion
    @Column(name = "markAsRemoved")
    private boolean markAsRemoved;
```

File: src/test/java/pvs/app/entity/ProjectTest.java [hunk 9]
```diff
@@ -0,0 +1,42 @@
...
+
+import static org.junit.Assert.assertEquals;
+
+@RunWith(SpringRunner.class)
+@SpringBootTest(classes = Application.class)
+public class ProjectTest {
+    Project myProject;
+    final String testProjectName = "myProject";
+    Integer defaultStatus = 1;
+    Integer deletedStatus = 0;
```
Comment (Xanonymous-GitHub) [inline src/test/java/pvs/app/entity/ProjectTest.java:18]:
```suggestion
    final String testProjectName = "myProject";
    final boolean defaultProjectRemoveStatus = false;
    Project givenProject;
```

File: src/test/java/pvs/app/entity/ProjectTest.java [hunk 10]
```diff
@@ -0,0 +1,42 @@
...
+
+    @Before
+    public void setup() {
+        myProject = new Project();
+        myProject.setMemberId(1L);
+        myProject.setName(testProjectName);
```
Comment (Xanonymous-GitHub) [inline src/test/java/pvs/app/entity/ProjectTest.java:24]:
```suggestion
        givenProject = new Project();
        givenProject.setName(testProjectName);
```

File: src/test/java/pvs/app/entity/ProjectTest.java [hunk 11]
```diff
@@ -0,0 +1,42 @@
...
+    }
+
+    @Test
+    public void getNameTest() {
+        assertEquals(testProjectName, myProject.getName());
```
Comment (Xanonymous-GitHub) [inline src/test/java/pvs/app/entity/ProjectTest.java:29]:
```suggestion
        assertEquals(testProjectName, givenProject.getName());
```

File: src/test/java/pvs/app/entity/ProjectTest.java [hunk 12]
```diff
@@ -0,0 +1,42 @@
...
+    }
```
Comment (Xanonymous-GitHub) [inline src/test/java/pvs/app/entity/ProjectTest.java:30]:
```suggestion
    public void correctlyGetProjectName() {
        assertEquals(testProjectName, givenProject.getName());
    }
```

File: src/test/java/pvs/app/entity/ProjectTest.java [hunk 13]
```diff
@@ -0,0 +1,42 @@
...
+
+    @Test
+    public void defaultOfStatusTest() {
+        assertEquals(defaultStatus, myProject.getStatus());
+    }
```
Comment (Xanonymous-GitHub) [inline src/test/java/pvs/app/entity/ProjectTest.java:35]:
```suggestion
    public void shouldHaveDefaultRemoveStatus() {
        assertEquals(defaultProjectRemoveStatus, givenProject.isMarkAsRemoved());
    }
```

File: src/test/java/pvs/app/entity/ProjectTest.java [hunk 14]
```diff
@@ -0,0 +1,42 @@
...
+
+    @Test
+    public void deleteMyProjectTest() {
+        myProject.setStatus(deletedStatus);
+        assertEquals(deletedStatus, myProject.getStatus());
+    }
```
Comment (Xanonymous-GitHub) [inline src/test/java/pvs/app/entity/ProjectTest.java:41]:
```suggestion
    public void correctlyMarkAsRemoved() {
        givenProject.setMarkAsRemoved(true);
        assertTrue(givenProject.isMarkAsRemoved());
    }
```

Comment (Xanonymous-GitHub) [reply src/test/java/pvs/app/entity/ProjectTest.java:41]: Addressed in eb5696a.

File: src/main/java/pvs/app/entity/Project.java (hunk 8 above)
Comment (VincentETN) [reply src/main/java/pvs/app/entity/Project.java:39]: If the getter of `boolean` porperty is prefixed "is" to the property name, I think that the property name could be just "removed". When we call the getter, `isRemoved()` is also readable.

Comment (VincentETN) [reply src/main/java/pvs/app/entity/Project.java]: 3f92894

File: src/test/java/pvs/app/entity/ProjectTest.java (hunk 9 above)
Comment (VincentETN) [reply src/test/java/pvs/app/entity/ProjectTest.java]: 7430f61

File: src/test/java/pvs/app/entity/ProjectTest.java (hunk 10 above)
Comment (VincentETN) [reply src/test/java/pvs/app/entity/ProjectTest.java]: 25d6e29

File: src/test/java/pvs/app/entity/ProjectTest.java (hunk 12 above)
Comment (VincentETN) [reply src/test/java/pvs/app/entity/ProjectTest.java]: b1ffccf

File: src/test/java/pvs/app/entity/ProjectTest.java (hunk 13 above)
Comment (VincentETN) [reply src/test/java/pvs/app/entity/ProjectTest.java]: b1ffccf

File: src/test/java/pvs/app/entity/ProjectTest.java (hunk 14 above)
Comment (VincentETN) [reply src/test/java/pvs/app/entity/ProjectTest.java]: b1ffccf

File: src/test/java/pvs/app/entity/ProjectTest.java (hunk 12 above)
Comment (Xanonymous-GitHub) [reply src/test/java/pvs/app/entity/ProjectTest.java]: Suggest rename to `removeAttributeShouldHaveDefaultValue`

File: src/test/java/pvs/app/entity/ProjectTest.java (hunk 13 above)
Comment (Xanonymous-GitHub) [reply src/test/java/pvs/app/entity/ProjectTest.java]: Suggest rename to `removeAttributeShouldHaveDefaultValue`

File: src/test/java/pvs/app/entity/ProjectTest.java (hunk 14 above)
Comment (Xanonymous-GitHub) [reply src/test/java/pvs/app/entity/ProjectTest.java]: Suggest rename to `removeAttributeShouldHaveDefaultValue`

**Review by jn2657 — APPROVED** [review]

File: src/test/java/pvs/app/entity/ProjectTest.java (hunk 14 above)
Comment (VincentETN) [reply src/test/java/pvs/app/entity/ProjectTest.java]: e219ac5

**Review by Xanonymous-GitHub — APPROVED** [review]
LGTM

**Review by hellenGUAN — APPROVED** [review]
